src/App.tsx

```diff
@@ -1,26 +1,163 @@
-/* eslint-disable max-len */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useMemo, useRef, useState } from 'react';
 import { UserWarning } from './UserWarning';
+import * as todoServices from './api/todos';
 
-const USER_ID = 0;
+import { Todo } from './types/Todo';
+import { FilterType } from './types/FilterType';
+import { ErrorType } from './types/ErrorType';
+
+import { Header } from './components/Header';
+import { TodoList } from './components/TodoList';
+import { Footer } from './components/Footer';
+import { ErrorNotification } from './components/ErrorNotification';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState<ErrorType>(
+    ErrorType.ERROR_DEFAULT,
+  );
+  const [filterTodoBy, setFilterTodoBy] = useState<FilterType>(FilterType.ALL);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [isLoading, setIsLoading] = useState(false);
+  const [deletedTodoId, setDeletedTodoId] = useState<number[] | null>(null);
+
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  //#region loadTodos and filteredTodos
+  useEffect(() => {
+    const asyncFetch = async () => {
+      inputRef.current?.focus();
+
+      try {
+        const loadTodos = await todoServices.getTodos();
+
+        setTodos(loadTodos);
+      } catch (error) {
+        setErrorMessage(ErrorType.ERROR_LOADING);
+        throw error;
+      }
+    };
+
+    asyncFetch();
+  }, []);
+
+  const filteredTodos = useMemo(() => {
+    return todos.filter(todo => {
+      if (FilterType.ACTIVE === filterTodoBy) {
+        return !todo.completed;
+      }
+
+      if (FilterType.COMPLETED === filterTodoBy) {
+        return todo.completed;
+      }
+
+      return true;
+    });
+  }, [todos, filterTodoBy]);
+  //#endregion
+
+  //#region addNewTodo
+  function addTodo({ id, userId, title, completed }: Todo) {
```

Comment:
It is important to stick to one way of creating functions, fix it everywhere
```suggestion
  const addTodo = ({ id, userId, title, completed }: Todo) => {
```